carbon-design-system/carbon
# fix(Toggletip): general a11y / ux improvements

Closes https://github.com/carbon-design-system/carbon/issues/12922

Addresses general UX / a11y issues found in the `Toggletip` component

#### Changelog

**New**

- `Toggletip` will now close when the menu is tabbed through, and focus leaves the popover body.
- If <kbd>Escape</kbd> is pressed while an item inside the `Toggletip` is focused, the focus will return to the `Toggletip` trigger 

**Changed**

- Updated `ToggletipLabel` to `body-01` to match the design specs
- Labelled the stories better to note that `Toggletip` is not normally rendered `open` by default

#### Testing / Reviewing

- Open the `Toggletip` story 
- <kbd>Tab</kbd> through the component. Focus should go to the trigger, then press <kbd>Space</kbd> to open the menu. <kbd>Tab</kbd> through the component, and the menu should close when the `focus` leaves the popover body.
-  <kbd>Tab</kbd> through the component. While an item inside has `focus` (not the trigger), hit the <kbd>Escape</kbd> key. The `focus` should return to the trigger.
- Ensure all existing functionality remains
